helmcharts/notes/templates/deployment.yaml

```diff
@@ -0,0 +1,86 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "notes.fullname" . }}
+  labels:
+{{ include "notes.labels" . | indent 4 }}
+spec:
+  replicas: {{ .Values.app.replicaCount }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "notes.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "notes.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+    {{- with .Values.app.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      # Create a volume with the application secrets
+      volumes:
+        - name: {{ include "notes.secretName" . }}
+          secret:
+            secretName: {{ include "notes.secretName" . }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.app.image.repository }}:{{ .Values.app.image.tag }}"
+          imagePullPolicy: {{ .Values.app.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: 8120
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /
+              port: http
+          readinessProbe:
+            httpGet:
+              path: /heartbeat
+              port: http
+          
+          # Expose application secret contents as a volume in /edx/etc
+          volumeMounts:
+            - mountPath: "/edx/etc"
+              name: {{ include "notes.secretName" . }}
+              readOnly: true
+          resources:
+            {{- toYaml .Values.app.resources | nindent 12 }}
+      initContainers:
+      - name: init-elasticsearch
+        image: busybox:1.29.3
```

Comment:
Just did some reading on the BusyBox image.  Would totally be worth a comment somewhere about what this image is and why it is useful.

Reply:
Added